extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTools.java

```diff
@@ -16,4 +22,72 @@ public static IcebergCatalogAdapter createAdapter(
             final FileIO fileIO) {
         return new IcebergCatalogAdapter(catalog, fileIO);
     }
+
+    /**
+     * <p>
+     * Create an Iceberg catalog adapter for an Iceberg catalog created from configuration properties. These properties
+     * map to the Iceberg catalog Java API properties and are used to create the catalog and file IO implementations.
+     * </p>
+     * <p>
+     * This is a wrapper around {@link CatalogUtil#buildIcebergCatalog(String, Map, Object)} and accepts the same
+     * properties. The minimal set of properties required to create an Iceberg catalog are:
+     * <ul>
+     * <li>{@code "catalog-impl"} or {@code "type"} - the Java catalog implementation to use. When providing
+     * {@code "catalog-impl"}, the implementing Java class should be provided (e.g.
+     * {@code "org.apache.iceberg.rest.RESTCatalog"} or {@code "org.apache.iceberg.aws.glue.GlueCatalog")}. Choices for
+     * {@code "type"} include {@code "hive"}, {@code "hadoop"}, {@code "rest"}, {@code "glue"}, {@code "nessie"},
+     * {@code "jdbc"}.</li>
+     * <li>{@code "uri"} - the URI of the catalog.</li>
+     * <li>{@code "io-impl"} - the Java FileIO implementation to use. Common choices are:
+     * {@code "org.apache.iceberg.aws.s3.S3FileIO"} and {@code "org.apache.iceberg.hadoop.HadoopFileIO"}</li>
+     * </ul>
+     * <p>
+     * Other common properties include:
+     * </p>
+     * <ul>
+     * <li>{@code "warehouse"} - the location of the data warehouse.</li>
+     * <li>{@code "client.region"} - the region of the AWS client.</li>
+     * <li>{@code "s3.access-key-id"} - the S3 access key for reading files.</li>
+     * <li>{@code "s3.secret-access-key"} - the S3 secret access key for reading files.</li>
+     * <li>{@code "s3.endpoint"} - the S3 endpoint to connect to.</li>
+     * </ul>
+     * <p>
+     * Additional properties for the specific catalog should also be included, such as as S3-specific properties for
+     * authentication or endpoint overriding.
+     * </p>
+     * 
+     * @param name the name of the catalog; if omitted, the catalog URI will be used to generate a name
+     * @param properties the map containing the Iceberg catalog properties to use
+     * @return the Iceberg catalog adapter
+     */
+    public static IcebergCatalogAdapter createAdapter(
+            @Nullable final String name,
+            @NotNull final Map<String, String> properties) {
+
+        // Validate the minimum required properties are set.
+        if (!properties.containsKey(CatalogProperties.CATALOG_IMPL) && !properties.containsKey("type")) {
+            throw new IllegalArgumentException(String.format("Catalog type or implementation property '%s' is required",
+                    CatalogProperties.CATALOG_IMPL));
```

Comment:
We might be able to infer the catalog type based on URI like pyiceberg does:
* https://py.iceberg.apache.org/reference/pyiceberg/catalog/#pyiceberg.catalog.infer_catalog_type
* https://github.com/apache/iceberg-python/blob/pyiceberg-0.6.1/pyiceberg/catalog/__init__.py#L155-L182

Reply:
We can do this and it will bring us closer to PyIceberg configuration.  My approach was not to repeat all the hand-holding that pyiceberg does, but to mirror the Java Iceberg API which is more barebones.

Not sure which approach is better, but I lean toward less code and more user responsibility (esp. where it matches Iceberg Java API).

Reply:
I'm OK with introducing these kinds of helpers, as long as we can be sure that they are:
1. Unambiguous
2. Unlikely to break later because of changes to the expected properties

That said, we can merge property-driven catalog support independently and add these kinds of "nice to have" features later; they will not change the interface, as far as I can tell.